include/settings/settings.h

```diff
@@ -42,8 +42,64 @@ typedef ssize_t (*settings_read_cb)(void *cb_arg, void *data, size_t len);
  * These are registered using a call to @ref settings_register.
  */
 struct settings_handler {
+
+	char *name;
```

Comment:
I would extract common structure part for static and dynamic handler (setting_handler_base/common) and then declare static and structure based on that.

function like settings_parse_and_lookup() might even use it as return value type.

Reply:
I tried doing something like that. The problem is that the node info is part of the dynamic handler, so you have to define the dynamic handler as `struct settings_handler test = {...}`
if you would make the dynamic handler as the combination of static handler + node, you would need to define the above as `struct settings_handler test = { .static.get = ...}` this would just seem strange, and it would be an API change for dynamic handlers.
I found the way it is now the simplest and cleanest.
BTW: I have changed the return type of `settings_parse_and_lookup()` to the static type.

Reply:
OK, I can live with how this is implemented.

Reply:
Although both struct have to be almost the same - If somebody changes only one of them, then will introduce a bug.

Reply:
> this would just seem strange, and it would be an API change for dynamic handlers.

Actually was thinking about making it symmetric.
```C
struct settings_handler_basic {
    char *name;
   ...
};

struct settings_handler_static {
    struct struct settings_handler_basic base;
};

struct settings_handler {
    struct settings_handler_basic base;
    sys_snode_t node;
    /**< Linked list node info for module internal usage. */
};

```

Reply:
I men I can accept API change. This is not critical so can resign from this, just let me know whether you think that no wort to do so.

Reply:
It would still be strange to create a dynamic settings_handler:
```
static struct settings_handler ps_handler = {
  .base.name = "ps",
  .base.h_get = get,
};
```
As the settings module is already complicated to use I would not do this. I think the next thing to do is to remove the dynamic handlers and only to support static ones.